### DISCUSSION TITLE
HTAB (\t) are not allowed in header values when using `httptools`

### DISCUSSION THREAD

**Discussion by jhominal** [Kludex/uvicorn#2344] [Potential Issue] [answered]
The regular expression to validate header values is [`[\x00-\x1F\x7F]`](https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/http/httptools_impl.py#L29)

However, [RFC 9110, section 5.5](https://www.rfc-editor.org/rfc/rfc9110#section-5.5) allows both spaces (`0x20`) and horizontal tabs (`0x09`). However, the regular expression above will erroneously detect `\x09` as being invalid.

For that reason, I think that the regular expression should be amended to allow horizontal tabs.

As a comparison basis, note that `h11` allows `\t` character in header values: https://github.com/python-hyper/h11/blob/cc87dfcc5a4693eb49b0453e6677ca004ffb035b/h11/_abnf.py#L57

**Comment by Kludex** [comment, marked as answer]
PR welcome

**Reply by jhominal** [reply]
#2345 :)
